README.md

```diff
@@ -38,21 +44,73 @@ import '@uppy/dashboard/dist/style.css'
 import Dashboard from '@uppy/dashboard'
 import ImageKitUppyPlugin from "imagekit-uppy-plugin"
 
+const authenticator = () => {
+    // must return a promise
+    return new Promise((resolve, reject) => {
+        //code to fetch security parameters for authentication
+    })
+}
+
 const uppy = Uppy({ debug: true, autoProceed: false })
     .use(Dashboard, {
         inline: true,
         trigger: '#uppyDashboard', // your element
     })
     .use(ImageKitUppyPlugin, {
         id: 'ImageKit',
-        authenticationEndpoint: `http://www.yourserver.com/auth`,
+        authenticator,
         publicKey: "your_public_key"
     })
 ```
 
-The plugin makes an HTTP GET request to `authenticationEndpoint` and expects a JSON response with three fields i.e. `signature`, `token` and `expire`. In addition, the plugin adds a query parameter t with a random value to ensure that the request URL is unique and the response is not cached in Safari iOS. Your backend can ignore this query parameter.
+The plugin utilizes an asynchronous function named `authenticator`, which is intended to be used for retrieving security parameters from your backend. This function is expected to resolve with an object containing three fields: `signature`, `token`, and `expire`.
+
+Example implementation for `authenticator` using `XMLHttpRequest`.
```

Comment:
We can give a better example of implementing authentication endpoint. One which shows usage of headers, publicKey etc to get authentication parameters.

Moreover, the below logic is not of much use for the developer.

```
if (url.indexOf("?") === -1) {
       url += `?t=${Math.random().toString()}`;
} else {
       url += `&t=${Math.random().toString()}`;
}
```

We had it in our internal implementation of authenticationEndpoint for a specific reason and it is not applicable now. 

Suggestion: Perhaps we can make a simpler and concise example using the fetch API.